smart_open/smart_open_lib.py

```diff
@@ -126,19 +127,27 @@ def smart_open(uri, mode="rb", **kw):
             # local files -- both read & write supported
             # compression, if any, is determined by the filename extension (.gz, .bz2)
             return file_smart_open(parsed_uri.uri_path, mode)
-        elif parsed_uri.scheme in ("s3", "s3n"):
+        elif parsed_uri.scheme in ("s3", "s3n", "s3u"):
+            kwargs = {}
             # Get an S3 host. It is required for sigv4 operations.
-            host = kw.pop('host', None)
-            if not host:
-                host = boto.config.get('s3', 'host', 's3.amazonaws.com')
+            host = kw.pop('host', parsed_uri.host)
+            port = kw.pop('port', parsed_uri.port)
```

Comment:
Does this mean host and port are now mandatory in URIs?

That wouldn't be an acceptable change (old behaviour must continue to work the same).